repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasRelationshipStoreV2.java

```diff
@@ -211,6 +212,35 @@ public AtlasRelationship update(AtlasRelationship relationship) throws AtlasBase
         return ret;
     }
 
+    @Override
+    @GraphTransaction
+    public List<AtlasRelationship> createOrUpdate(List<AtlasRelationship> relationships) throws AtlasBaseException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("==> createOrUpdate({})", relationships);
+        }
+        List<AtlasRelationship> ret = new ArrayList<>();
+
+        for (AtlasRelationship relationship : relationships) {
+            AtlasVertex end1Vertex = getVertexFromEndPoint(relationship.getEnd1());
+            AtlasVertex end2Vertex = getVertexFromEndPoint(relationship.getEnd2());
+            String relationshipLabel = getRelationshipEdgeLabel(end1Vertex, end2Vertex, relationship.getTypeName());
+
```

Comment:
Can you add NPE check for end1Vertex & end2Vertex ?  If vertex is already there can be NPE.